src/service/document-info-impl.js

```diff
@@ -96,6 +103,10 @@ export class DocInfo {
       get sourceUrl() {
         return getSourceUrl(ampdoc.getUrl());
       },
+      /** @return {!Promise<string>} */
+      get pageViewId64() {
```

Comment:
I do believe you don't want to calculate a new id every time it is retrieved. Please calculate the value first and return the cached value in getter. Thanks